crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -161,6 +160,22 @@ module std::vector {
 
     // === Macros ===
 
+    /// Create a vector of length `n` by calling the function `f` on each index.
+    public macro fun tabulate<$T>($n: u64, $f: |u64| -> $T): vector<$T> {
+        let mut v = vector[];
+        let n = $n;
+        n.do!(|i| v.push_back($f(i)));
+        v
+    }
+
+    /// Count how many elements in the vector `v` satisfy the predicate `f`.
+    public macro fun count<$T>($v: &vector<$T>, $f: |&$T| -> bool): u64 {
+        let v = $v;
+        let mut count = 0;
+        v.do_ref!(|e| if ($f(e)) count = count + 1);
+        count
+    }
```

Comment:
This is like a special version of `do_ref` or `fold`. I would probably put it after one of those two sections. That is either the `do*` or after `fold`

Reply:
You mean ordering in the module scope? Sure thing.